public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiHealth,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { get, isEmpty } from 'lodash';
+import { RouteComponentProps } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHideSideNavBar } from '../../../main/hooks/useHideSideNavBar';
+import { Listener } from '../../../../utils/utils';
+import { darkModeEnabled } from '../../../../utils/kibanaUtils';
+import { AppState } from '../../../../redux/reducers';
+import {
+  startDetector,
+  stopDetector,
+  deleteDetector,
+  getDetector,
+  getHistoricalDetectorList,
+} from '../../../../redux/reducers/ad';
+import { BREADCRUMBS } from '../../../../utils/constants';
+import { DETECTOR_STATE } from '../../../../../server/utils/constants';
+import { HistoricalDetectorConfig } from '../../components/HistoricalDetectorConfig/HistoricalDetectorConfig';
+import { HistoricalDetectorControls } from '../../components/HistoricalDetectorControls/HistoricalDetectorControls';
+import { EditHistoricalDetectorModal } from '../ActionModals/EditHistoricalDetectorModal/EditHistoricalDetectorModal';
+import { DeleteHistoricalDetectorModal } from '../ActionModals/DeleteHistoricalDetectorModal/DeleteHistoricalDetectorModal';
+import { AnomalyHistory } from '../../../DetectorResults/containers/AnomalyHistory';
+import { stateToColorMap } from '../../../utils/constants';
+import {
+  HISTORICAL_DETECTOR_RESULT_REFRESH_RATE,
+  HISTORICAL_DETECTOR_ACTION,
+  HISTORICAL_DETECTOR_STOP_THRESHOLD,
+} from '../../utils/constants';
+import {
+  getCallout,
+  waitForMs,
+  getRunningHistoricalDetectorCount,
+} from '../../utils/helpers';
+import { CoreStart } from '../../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../../components/CoreServices/CoreServices';
+import { getClusterStats } from '../../../../redux/reducers/elasticsearch';
+import { GET_ALL_DETECTORS_QUERY_PARAMS } from '../../../utils/constants';
+
+export interface HistoricalDetectorRouterProps {
+  detectorId?: string;
+}
+interface HistoricalDetectorDetailProps
+  extends RouteComponentProps<HistoricalDetectorRouterProps> {}
+
+interface HistoricalDetectorDetailModalState {
+  isOpen: boolean;
```

Comment:
What does `isOpen` mean?

Reply:
`isOpen` just means if the modal is open or not. Further down in this file is the `getHistoricalDetectorDetailModal` function with uses the modal state to determine what, if any, modal should currently render.

Reply:
Got it